File: src/web/magmac/app/lang/TypeSeparatorFolder.ts
export class TypeSeparatorFolder {
	public fold( state : DivideState,  c : char) : DivideState {if(true){ return state.advance( );;} let appended : var=state.append( c);if(true){ return appended.enter( );;}if(true){ return appended.exit( );;}return appended;;}
	public createDelimiter() : String {return " ";;}
}

File: src/java/magmac/app/lang/TypeSeparatorFolder.java
package magmac.app.lang;

import magmac.app.compile.rule.divide.DivideState;
import magmac.app.compile.rule.fold.Folder;

public class TypeSeparatorFolder implements Folder {
    @Override
    public DivideState fold(DivideState state, char c) {
        if (' ' == c && state.isLevel()) {
            return state.advance();
        }

        var appended = state.append(c);
        if ('<' == c) {
            return appended.enter();
        }
        if ('>' == c) {
            return appended.exit();
        }
        return appended;
    }

    @Override
    public String createDelimiter() {
        return " ";
    }
}
